comps/retrievers/langchain/vdms/retriever_vdms.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import time
+
+from langchain_community.embeddings import HuggingFaceBgeEmbeddings, HuggingFaceHubEmbeddings
+from langchain_community.vectorstores.vdms import VDMS, VDMS_Client
+from langchain_huggingface.embeddings import HuggingFaceEndpointEmbeddings
+from langsmith import traceable
+from vdms_config import (  # , HUGGINGFACEHUB_API_TOKEN, INDEX_SCHEMA, VDMS_URL
+    DISTANCE_STRATEGY,
+    EMBED_MODEL,
+    INDEX_NAME,
+    SEARCH_ENGINE,
+    VDMS_HOST,
+    VDMS_PORT,
+)
+
+from comps import (
+    EmbedDoc,
+    SearchedDoc,
+    SearchedMultimodalDoc,
+    ServiceType,
+    TextDoc,
+    opea_microservices,
+    register_microservice,
+    register_statistics,
+    statistics_dict,
+)
+
+tei_embedding_endpoint = os.getenv("TEI_EMBEDDING_ENDPOINT")
+hf_token = os.getenv("HUGGINGFACEHUB_API_TOKEN")
+use_clip = int(os.getenv("USECLIP"))
+
+if use_clip:
+    import sys
+
+    sys.path.append("../../../embeddings/langchain_multimodal/")
+    from embeddings_clip import vCLIP
+
+# Debugging
+all_variables = dir()
+
+for name in all_variables:
+    # Print the item if it doesn't start with '__'
+    if not name.startswith("__"):
+        myvalue = eval(name)
+        print(name, "is", type(myvalue), "and = ", myvalue)
+
+
+client = VDMS_Client(VDMS_HOST, VDMS_PORT)
+
+
+@register_microservice(
+    name="opea_service@retriever_vdms",
+    service_type=ServiceType.RETRIEVER,
+    endpoint="/v1/retrieval",
+    host="0.0.0.0",
+    port=7000,
+)
+@traceable(run_type="retriever")
```

Comment:
Please remove the langsmith related code

Reply:
Done